long-ride-alerts/src/test/java/org/apache/flink/training/exercises/longrides/LongRidesHarnessTest.java

```diff
@@ -0,0 +1,56 @@
+package org.apache.flink.training.exercises.longrides;
+
+import org.apache.flink.api.common.typeinfo.BasicTypeInfo;
+import org.apache.flink.streaming.api.operators.KeyedProcessOperator;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.util.KeyedOneInputStreamOperatorTestHarness;
+import org.apache.flink.training.exercises.common.datatypes.TaxiRide;
+import org.apache.flink.training.solutions.longrides.LongRidesSolution;
+
+import org.junit.Test;
+
+import java.util.concurrent.ConcurrentLinkedQueue;
+
+import static org.assertj.core.api.AssertionsForInterfaceTypes.assertThat;
+
+public class LongRidesHarnessTest {
+
+    @Test
+    public void testLongRideAlertsAsSoonAsPossible() throws Exception {
```

Comment:
I think, we'd need a couple of more tests to verify the full behaviour, including:

- alerting directly if you see the end event first
- not alerting for a start event until the watermark is exactly 2h ahead (even if you see later data or processing time advances beyond 2h)
- not alerting for an end event that is not too long
- alerting if there is no event but the watermark surpasses the 2h
- state cleanup (in the good case - not the leak we leave for the open discussion)

Should we also verify that things, i.e. the start events, are kept in Flink state after all and not some other local Java store (best-effort by just looking at the number of state entries > 0)?